### PR TITLE
new debt ratio metric

### PR DESCRIPTION
**Pull request by sns-seb**
This PR adds the computation of metric `new_squale_debt_ratio` to the CE

### CODE REVIEW

**Comment by julienlancelot** [comment]
Sounds good, except missing unit tests :(

**Comment by sns-seb** [comment]
@julienlancelot added unit test, rebased on master and applied other feedback

**Comment by julienlancelot** [comment]
@seb Good job ! Just some missing ut and LGTM.

File: server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+      return 0L;
+    }
+    return getLongValue(measure.get(), period);
+  }
+
+  private static long getLongValue(Measure measure, Period period) {
+    if (measure.hasVariations() && measure.getVariations().hasVariation(period.getIndex())) {
+      return (long) measure.getVariations().getVariation(period.getIndex());
+    }
+    return 0l;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java]: MAJOR Replace this lower case "l" long suffix by an upper case "L". [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ALowerCaseLongSuffixCheck)
MINOR Upper-case this literal "l" suffix. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS818)

File: server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+      measureRepository.add(component, this.newDebtRatioMetric, measure);
+    }
+  }
+
+  private double computeDensity(Component component, Period period, long developmentCost) {
+    double debt = getLongValue(measureRepository.getRawMeasure(component, this.newDebtMetric), period);
+    if (developmentCost != 0L) {
+      return debt / (double) developmentCost;
+    }
+    return 0d;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java]: MINOR Upper-case this literal "d" suffix. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS818)

File: server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+  }
+
+  private static long getLongValue(Measure measure, Period period) {
+    if (measure.hasVariations() && measure.getVariations().hasVariation(period.getIndex())) {
+      return (long) measure.getVariations().getVariation(period.getIndex());
+    }
+    return 0l;
+  }
+
+  private void initNewDebtRatioCounter(Component file, Path<NewDevelopmentCostCounter> path) {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/sqale/SqaleNewMeasuresVisitor.java]: MAJOR The Cyclomatic Complexity of this method "initNewDebtRatioCounter" is 11 which is greater than 10 authorized. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Comment by sns-seb** [comment]
convered the last bits of `SqaleNewMeasuresVisitor`, will merge as soon as build on travis is green

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 2 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

**Pull request merged**